coderd/inboxnotifications.go

```diff
@@ -28,16 +29,63 @@ const (
 	notificationFormatPlaintext = "plaintext"
 )
 
+const (
+	FallbackIconWorkspace = "DEFAULT_WORKSPACE_ICON"
+	FallbackIconAccount   = "DEFAULT_ACCOUNT_ICON"
+	FallbackIconTemplate  = "DEFAULT_TEMPLATE_ICON"
+	FallbackIconOther     = "DEFAULT_OTHER_ICON"
+)
+
+var fallbackIcons = map[uuid.UUID]string{
+	// workspace related notifications
+	notifications.TemplateWorkspaceCreated:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManuallyUpdated:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDeleted:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutobuildFailed:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDormant:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutoUpdated:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceMarkedForDeletion: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManualBuildFailed: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfMemory:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfDisk:         FallbackIconWorkspace,
+
+	// account related notifications
+	notifications.TemplateUserAccountCreated:           FallbackIconAccount,
+	notifications.TemplateUserAccountDeleted:           FallbackIconAccount,
+	notifications.TemplateUserAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateUserAccountActivated:         FallbackIconAccount,
+	notifications.TemplateYourAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateYourAccountActivated:         FallbackIconAccount,
+	notifications.TemplateUserRequestedOneTimePasscode: FallbackIconAccount,
+
+	// template related notifications
+	notifications.TemplateTemplateDeleted:             FallbackIconTemplate,
+	notifications.TemplateTemplateDeprecated:          FallbackIconTemplate,
+	notifications.TemplateWorkspaceBuildsFailedReport: FallbackIconTemplate,
+}
+
+func SetInboxNotificationIcon(notif *codersdk.InboxNotification) {
```

Comment:
I don't think there is a need to expose this function outside. It is rather an internal method to take default icon if not present.

```go
notif = ensureNotificationIcon(notif)
```

```
func ensureNotificationIcon(notif codersdk.InboxNotification) {
     // 
}
```

PS. mind the original `notif` stays unmodified - idempotency.